### PR TITLE
Add hash values to tags in cirq.google

### PR DESCRIPTION
**Pull request by dstrain115**
- By mypy definitions, tags should be hashable.
- Add __hash__ fucntions so they actually are hashable.

### CODE REVIEW

**Review by mpharrigan — APPROVED** [review]
what do you mean "according to mypy"? Is there a test that can catch these issues automatically?

File: cirq/google/ops/physical_z_tag.py [hunk 1]
```diff
@@ -42,3 +42,6 @@ def _json_dict_(self) -> Dict[str, Any]:
 
     def __eq__(self, other) -> bool:
         return isinstance(other, PhysicalZTag)
+
+    def __hash__(self) -> int:
+        return 123
```
Comment (vtomole) [inline cirq/google/ops/physical_z_tag.py:47]: Why `123`?

**Comment by dstrain115** [comment]
> what do you mean "according to mypy"? Is there a test that can catch these issues automatically?

We don't have a sub-class for Tags, we just call them Hashable in the with_tags.  I am not sure if there is a way to test this in general or not.
https://github.com/quantumlib/Cirq/blob/master/cirq/ops/raw_types.py#L428

I feel like we have used with_tags(cirq.google.PhysicalZ()) elsewhere and mypy didn't complain that it wasn't Hashable though.

File: cirq/google/ops/physical_z_tag.py (hunk 1 above)
Comment (dstrain115) [reply cirq/google/ops/physical_z_tag.py:47]: Just an arbitrary value so that all instances hash to the same value.

**Review by maffoo — APPROVED** [review]
LGTM

**Pull request merged**